### PR TITLE
HDDS-3747. Redundancy if condition code in ListPipelinesSubcommand

### PR DESCRIPTION
**Pull request by hemanthboyina**
Remove the redundancy if condition code in ListPipelinesSubcommand

## What changes were proposed in this pull request?

to remove redundancy code in ListPipelinesSubCommand

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-3747

## How was this patch tested?

through CI checks

### CODE REVIEW

File: hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/pipeline/ListPipelinesSubcommand.java [hunk 1]
```diff
@@ -54,17 +54,12 @@
...
-                (p.getPipelineState().toString().compareToIgnoreCase(state)
-                    == 0))))
-            .forEach(System.out::println);
-      }
+      scmClient.listPipelines().stream()
+          .filter(p -> ((Strings.isNullOrEmpty(factor)
+              || (p.getFactor().toString().compareToIgnoreCase(factor) == 0))
+              && (Strings.isNullOrEmpty(state) || (p.getPipelineState()
+                  .toString().compareToIgnoreCase(state) == 0))))
+          .forEach(System.out::println);
```
Comment (adoroszlai) [inline hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/pipeline/ListPipelinesSubcommand.java:62]: Structuring this as below would reduce duplication, and make it easier to add further filters if needed:

```
Stream<...> stream = scmClient.listPipelines().stream();
if (!isNullOrEmpty(factor)) {
  stream = stream.filter(...);
}
if (!isNullOrEmpty(state)) {
  stream = stream.filter(...);
}
stream.forEach(...);
```

**Comment by bshashikant** [comment]
@bhemanthkumar , can you please fill up the PR details as requested in the template?

File: hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/pipeline/ListPipelinesSubcommand.java (hunk 1 above)
Comment (maobaolong) [reply hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/pipeline/ListPipelinesSubcommand.java:62]: @bhemanthkumar i think the changes within the try block @adoroszlai is
```java
      Stream<Pipeline> stream = scmClient.listPipelines().stream();
      if (!Strings.isNullOrEmpty(factor)) {
        stream =
            stream.filter(p -> p.getFactor().toString()
                .compareToIgnoreCase(factor) == 0);
      }
      if (!Strings.isNullOrEmpty(state)) {
        stream =
            stream.filter(p -> p.getPipelineState().toString()
                .compareToIgnoreCase(state) == 0);
      }
      stream.forEach(System.out::println);
```

**Comment by maobaolong** [comment]
@dineshchitlangia Would you like to merge this PR for @bhemanthkumar ?

**Comment by hemanthboyina** [comment]
@dineshchitlangia can we push this forward

**Pull request merged**